### PR TITLE
Add Cypress machine listing tests.

### PR DESCRIPTION
**Pull request by squidsoup**
## Done
- Add initial machine listing cypress tests.
- Bypass 'intro' for cypress tests by setting `skipintro` cookie.
- Move `getCookie` to `app/utils` (and migrate to ts).

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- n/a tests should pass.

## Fixes

Fixes: canonical-web-and-design/MAAS-squad#1985

### CODE REVIEW

File: proxy/cypress/integration/machines/list.spec.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { customAlphabet } from 'nanoid'
+
+import { generateMac, login } from "../utils";
+
+const nanoid = customAlphabet('1234567890abcdefghi', 10);
```
Comment (squidsoup) [inline proxy/cypress/integration/machines/list.spec.js:5]: nanoid generates ids with `_` by default which are invalid for `machine.hostname`.

File: proxy/cypress/integration/machines/list.spec.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+context("Machine listing", () => {
+  beforeEach(() => {
+    Cypress.config('defaultCommandTimeout', 10000);
+    login();
+    cy.setCookie('skipintro', 'true');
```
Comment (huwshimi) [inline proxy/cypress/integration/machines/list.spec.js:10]: Does this need to be cleaned up after each test?

File: proxy/cypress/integration/machines/list.spec.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    cy.visit(
+      `${Cypress.env("BASENAME")}${Cypress.env("REACT_BASENAME")}/machines`
```
Comment (huwshimi) [inline proxy/cypress/integration/machines/list.spec.js:12]: Doesn't have to be in this branch but we should definitely create a util for generating these URLs.

File: proxy/cypress/integration/machines/list.spec.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { customAlphabet } from 'nanoid'
+
+import { generateMac, login } from "../utils";
+
+const nanoid = customAlphabet('1234567890abcdefghi', 10);
+context("Machine listing", () => {
+  beforeEach(() => {
+    Cypress.config('defaultCommandTimeout', 10000);
```
Comment (huwshimi) [inline proxy/cypress/integration/machines/list.spec.js:8]: Why did you have to do this? Might be good to leave a comment.

File: proxy/cypress/integration/machines/list.spec.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    cy.visit(
+      `${Cypress.env("BASENAME")}${Cypress.env("REACT_BASENAME")}/machines`
+    );
+  });
+
+  it("renders a heading", () => {
+    cy.get("li.p-heading--four").contains("Machines");
+  });
+
+  it("can add a machine", () => {
```
Comment (huwshimi) [inline proxy/cypress/integration/machines/list.spec.js:20]: Should this be in a different file as it's about a different view?

Comment (squidsoup) [reply proxy/cypress/integration/machines/list.spec.js]: done

**Pull request merged**